example/debug/odp_debug.c

```diff
@@ -260,21 +271,70 @@ static int queue_debug(void)
 	queue = odp_queue_create(name, &param);
 
 	if (queue == ODP_QUEUE_INVALID) {
-		printf("Queue create failed: %s\n", name);
+		ODPH_ERR("Queue create failed: %s\n", name);
 		return -1;
 	}
 
 	printf("\n");
 	odp_queue_print(queue);
 
 	if (odp_queue_destroy(queue)) {
-		printf("Queue destroy failed: %s\n", name);
+		ODPH_ERR("Queue destroy failed: %s\n", name);
 		return -1;
 	}
 
 	return 0;
 }
 
+static int pktio_debug(void)
+{
+	odp_pool_t pool;
+	odp_pool_param_t pool_param;
+	odp_pktio_t pktio;
+	int pkt_len = 100;
+
+	odp_pool_param_init(&pool_param);
+	pool_param.type = ODP_POOL_PACKET;
+	pool_param.pkt.num = 10;
+	pool_param.pkt.len = pkt_len;
+
+	pool = odp_pool_create("debug_pktio_pool", &pool_param);
+
+	if (pool == ODP_POOL_INVALID) {
+		ODPH_ERR("Pool create failed\n");
+		return -1;
+	}
+
+	pktio = odp_pktio_open("loop", pool, NULL);
+
+	if (pktio == ODP_PKTIO_INVALID) {
+		ODPH_ERR("Pktio open failed\n");
+		return -1;
+	}
+
+	odp_pktio_print(pktio);
+
+	if (odp_pktio_close(pktio)) {
+		ODPH_ERR("Pktio close failed\n");
+		return -1;
+	}
+
+	if (odp_pool_destroy(pool)) {
+		ODPH_ERR("Pool destroy failed\n");
+		return -1;
+	}
+
+	return 0;
+}
+
+static int ipsec_debug(void)
+{
+	printf("\n");
+	odp_ipsec_print();
+
```

Comment:
When have also`odp_ipsec_sa_print()`, but support for this can be added in a separate commit.

Reply:
Yes, left that for the next round. It needs considerable number of lines (to check capabilities) to successfully create a SA of any supported type.